queries/typescript.core.scm

```diff
@@ -229,3 +230,65 @@
     (predefined_type)
   ] @type @_.leading.end.startOf
 ) @_.domain
+
+;;!! abstract class MyClass {}
+;;!  ^^^^^^^^^^^^^^^^^^^^^^^^
+(
+  (abstract_class_declaration) @class
+  (#not-parent-type? @class export_statement)
+)
+
+;;!! export abstract class MyClass {}
+;;!  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+(export_statement
+  (abstract_class_declaration)
+) @class
+
+;;!! class MyClass {}
+;;!        ^^^^^^^
+;;!  ----------------
+(abstract_class_declaration
+  name: (_) @name
+) @_.domain
+
+;;!! interface Type { name: string; }
+;;!                   ^^^^
+;;!                   xxxxxx
+;;!                   ------------
+(property_signature
+  name: (_) @collectionKey @_.removal.start
+  type: (_
+    ":"
+    (_) @_.removal.end.startOf
```

Comment:
As above, why not use leading and trailing?

Reply:
fixing